### ISSUE TITLE
Some very simple aql query stuck and can not be kill

### ISSUE THREAD

**Issue by reetahuang** [arangodb/arangodb#14184]
## My Environment

* __ArangoDB Version__:        3.7.10
* __Storage Engine__:             RocksDB
* __Deployment Mode__:       Cluster
* __Deployment Strategy__:   ArangoDB Starter
* __Configuration__:               [query] max-runtime = 3600
* __Infrastructure__:               own
* __Operating System__:        Centos 7.7
* __CPU core__:                      16
* __Total RAM in your machine__:        32Gb
* __Disks in use__:        HDD
* __Used Package__:    RedHat .rpm

__AQL query__:

> for doc in @@schema limit @from,@size return doc
@schema: Address
from: 0
size: 10

__AQL explain__:

> Query String (48 chars, cacheable: true):
 for doc in @@schema limit @from,@size return doc
Execution plan:
 Id   NodeType                  Site     Est.   Comment
  1   SingletonNode             DBS         1   * ROOT
  2   EnumerateCollectionNode   DBS   1166800     - FOR doc IN Address   /* full collection scan, 9 shard(s) */
  7   RemoteNode                COOR  1166800       - REMOTE
  8   GatherNode                COOR  1166800       - GATHER   /* parallel, unsorted */
  3   LimitNode                 COOR       10       - LIMIT 0, 10
  4   ReturnNode                COOR       10       - RETURN doc
Indexes used:
 none
Optimization rules applied:
 Id   RuleName
  1   scatter-in-cluster
  2   remove-unnecessary-remote-scatter
  3   parallelize-gather

__Problem__:
Some very simple aql query stuck and can not be kill，and coordinator service rase cpu to 100%
When i use the same aql to query，it can be successful very quick

![image](https://user-images.githubusercontent.com/31846800/118108810-0408dc80-b413-11eb-9ad0-cc5f0c77e552.png)
![image](https://user-images.githubusercontent.com/31846800/118112183-3ddbe200-b417-11eb-8942-13101cf24b96.png)
![image](https://user-images.githubusercontent.com/31846800/118113886-6f55ad00-b419-11eb-987a-44a8ebfc06ec.png)

curl -X GET "http://192.168.1.204:8529/_db/_system/_api/query/current?all=true" -H "Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJhcmFuZ29kYiIsInNlcnZlcl9pZCI6ImZvbyJ9.m5EuCvys0qSxgEgluxpkQCV-R1vI972gWXFeHW56AlU"
> {
	"id": "40682046411817",
	"database": "ningbo",
	"user": "haizhi",
	"query": "for doc in @@schema limit @from,@size return doc",
	"bindVars": {
		"@schema": "Address",
		"size": 10,
		"from": 0
	},
	"started": "2021-05-13T10:55:40Z",
	"runTime": 2.3555935416655615e+4,
	"state": "killed",
	"stream": false
},
{
	"id": "40682046411799",
	"database": "major_bigdata_tmp",
	"user": "haizhi",
	"query": "for doc in @@schema limit @from,@size return doc",
	"bindVars": {
		"@schema": "baiwan_normal_edge_table",
		"size": 10,
		"from": 0
	},
	"started": "2021-05-13T10:55:15Z",
	"runTime": 2.358090208447911e+4,
	"state": "killed",
	"stream": false
},
{
	"id": "40682046341310",
	"database": "major_bigdata_tmp",
	"user": "haizhi",
	"query": "for doc in @@schema limit @from,@size return doc",
	"bindVars": {
		"@schema": "baiwan_normal_edge_table",
		"size": 10,
		"from": 0
	},
	"started": "2021-05-12T17:07:25Z",
	"runTime": 8.765091985967197e+4,
	"state": "killed",
	"stream": false
},
{
	"id": "40682046412387",
	"database": "graph_police_auto_perm",
	"user": "haizhi",
	"query": "for doc in @@schema limit @from,@size return doc",
	"bindVars": {
		"@schema": "Company",
		"size": 10,
		"from": 0
	},
	"started": "2021-05-13T11:03:47Z",
	"runTime": 2.3069417330706492e+4,
	"state": "killed",
	"stream": false
},
{
	"id": "40682046416035",
	"database": "test_wanke01_tmp",
	"user": "haizhi",
	"query": "for doc in @@schema limit @from,@size return doc",
	"bindVars": {
		"@schema": "Company",
		"size": 10,
		"from": 0
	},
	"started": "2021-05-13T11:50:39Z",
	"runTime": 2.025728802682087e+4,
	"state": "killed",
	"stream": false
},
{
	"id": "41781558021913",
	"database": "ningbo",
	"user": "haizhi",
	"query": "for doc in @@schema limit @from,@size return doc",
	"bindVars": {
		"@schema": "Address",
		"size": 10,
		"from": 0
	},
	"started": "2021-05-13T11:28:06Z",
	"runTime": 2.1610508506584913e+4,
	"state": "killed",
	"stream": false
}

arangodb-debug.sh analyze --server.ask-jwt-secret
>Good: Your cluster does not have any zombies
Good: Your cluster does not have any zombie coordinators
Good: Your cluster does not have any zombie analyzer revisions
Good: Your cluster does not have any cleaned servers for failover
Good: Your cluster does not have broken collections
Good: Your cluster does not have any leftover collections from deleted databases
Good: Your cluster does not have any collections without shards
Good: Your cluster does not miss any collections used as leaders in distributeShardsLike
Good: Your cluster does not have any leaders placed on failed DBServers
Good: Your cluster does not have any followers placed on failed DBServers
Good: Your cluster does not have any dead primaries in Current
Good: Your cluster does not have any skeleton databases (databases without collections)
Good: Your cluster is not missing relevant system collections
Good: Your cluster does not have collections where followers are out of sync
Good: Your cluster does not have any collections with dead leader and no insync follower
Good: Your cluster does not have broken edge indexes
Good: Your cluster does correct sharding strategies
Good: Your cluster does not have any unplanned failover candidates
Good: Your cluster has consistent DOCUMENT calls
Good: Did not detect any issues in your cluster

__Expected result__:
fix this problem

**Comment by reetahuang** [comment]
i have add the configure query.max-runtime = 3600 in arangod.conf

**Comment by Simran-B** [comment]
@reetahuang Can you please upgrade to 3.7.11 and check if the problem persists? It contains at least one fix regarding query killing. From the CHANGELOG:

> Fix BTS-346: Improved handling of AQL query kill command in unlikely places,  before the query starts to execute and after the query is done but the result is still being written. Now the cleanup of queries works more reliably. This unreliable kill time windows were very short and unlikely to hit, although if one was hit transactions were not aborted, and collection locks could be lingering until query timeout.

Thanks!

**Comment by reetahuang** [comment]
@Simran-B I have upgrade to version 3.7.11.
Sadly the problem persists,and the stuck query can be disappeared only when the coordinator service is restarted.
Please let me know if I can provide you with any further information.

**Comment by jtfidje** [comment]
We're having very similar problems with our cluster. What I've found is that the queries get "stuck" when the following two statements are true:
- `LIMIT` is less than the number of documents in the collection
- `fullCount` option is set to `true

I've written a description in markdown that two of my teammates used to successfully reproduce the issue on their machines. It includes a docker-compose file to set up the cluster, so for convenience I'll link a PasteBin to it: https://pastebin.com/MQYMbSGD

**Comment by chriskvik** [comment]
@reetahuang we have a similar issue, could you verify that our "solution" mentioned in #14214 works for you aswell?

**Comment by reetahuang** [comment]
@chriskvik 
Thanks for your reply, we have some different situations:
- The number of documents in the collection is more than LIMIT, they are about millons 
- It's hardly to reappearance the situation of queries get stuck, when i found a stuck query and repeat it over and over, the
resault was normal

Now I am using a shell script restart the coordinator service at regular time

**Comment by mpoeter** [comment]
This issue seems to have the same root cause as #14214. This should be fixed via #14268 and #14270 (for 3.7). Until a release with the fix is available you can work around this problem by disabling the parallelize-gather rule for the query:
```
db._query({
  query: 'FOR some_object in some_collection LIMIT 10 RETURN some_object',
  options: {
    fullCount: true,
    optimizer: { rules: ["-parallelize-gather"] }
  }
})
```

**Comment by reetahuang** [comment]
@mpoeter Thanks for your reply, I'll try it.

**Comment by maxkernbach** [comment]
Hi @reetahuang,

Did you already have a chance to test the workaround?

A fix, which is most likely resolving the stuck queries, is available in the latest versions 3.7.13 and 3.7.14.
Could you please also give this a try?

**Comment by Simran-B** [comment]
@reetahuang Do the latest versions resolve the problem?
